.changelog/3910.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:improvement
+plugins/k8s: Add `prune_whitelist` for `kubernetes-apply`
```

Comment:
Did this make it into `feature/k8s-task-runner-improvements`?  I see the change in #3868 but noticed that you have a changelog entry here, and mentioned it in the PR description.